openedx/core/djangoapps/credit/migrations/0008_auto__del_index_creditprovider_provider_id__chg_field_crediteligibilit.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import SchemaMigration
+from django.db import models
+
+
+class Migration(SchemaMigration):
+
+    def forwards(self, orm):
+        # Changing field 'CreditEligibility.provider'
+        db.alter_column('credit_crediteligibility', 'provider_id', self.gf('django.db.models.fields.related.ForeignKey')(null=True, to=orm['credit.CreditProvider']))
+
+    def backwards(self, orm):
+        # Changing field 'CreditEligibility.provider'
+        db.alter_column('credit_crediteligibility', 'provider_id', self.gf('django.db.models.fields.related.ForeignKey')(default=1, to=orm['credit.CreditProvider']))
```

Comment:
Since for the backward migration we have to add some default value because this was not a nullable field. So I set it to `1`. 
@wedaly @feanil  adding a migration to make `provider` foreignKey nullable was not sufficient.

South was confused about removing index for `provider_id` field. Since I believe for `provider` foreignKey in `CreditEligibility` model, south was creating `provide_id` field in db with automatic indexing which was same as `provide_id` field in `CreditProvider` model (with manual index).
I have updated default migration file to stop the south from removing index (which was unnecessary).